pkg/query/flamegraph_arrow.go

```diff
@@ -0,0 +1,158 @@
+package query
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/apache/arrow/go/v10/arrow"
+	"github.com/apache/arrow/go/v10/arrow/array"
+	"github.com/apache/arrow/go/v10/arrow/memory"
+	"github.com/polarsignals/frostdb/pqarrow/builder"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/parca-dev/parca/pkg/profile"
+)
+
+const (
+	flamegraphFieldMappingStart   = "mapping_start"
```

Comment:
I wonder if we can make this code cleaner by having a map from field name to some struct that encapsulates the arrow type and holds a reference to the builder.